runtime/src/bank/check_transactions.rs

```diff
@@ -97,18 +99,24 @@ impl Bank {
             .zip(lock_results)
             .map(|(tx, lock_res)| match lock_res {
                 Ok(()) => {
+                    let feature_set: &FeatureSet = &self.feature_set;
+                    let transaction_fee = calculate_transaction_fee(
+                        tx.borrow(),
+                        self.fee_structure.lamports_per_signature,
+                        FeeFeatures::from(feature_set),
+                    );
                     let compute_budget_and_limits = process_compute_budget_instructions(
                         tx.borrow().program_instructions_iter(),
-                        &self.feature_set,
+                        feature_set,
                     )
                     .map(|limit| {
                         if let Some(compute_budget) = self.compute_budget {
                             // This block of code is only necessary to retain legacy behavior of the code.
                             // It should be removed along with the change to favor transaction's compute budget limits
                             // over configured compute budget in Bank.
-                            compute_budget.get_compute_budget_and_limits(&limit)
+                            compute_budget.get_compute_budget_and_limits(&limit, transaction_fee)
                         } else {
-                            limit.get_compute_budget_and_limits()
+                            limit.get_compute_budget_and_limits(transaction_fee)
```

Comment:
We could just pass the fee details here it seems?

Reply:
The `prioritization_fee` is calculated inside the function. We can move the full function logic here (not sure if that's what you suggested).

Reply:
Or, maybe just move FeeBudgetLimits::from() out of the `get_compute_budget_and_limits()`. I am prototyping it. If it looks clean, I'll push an update.

Reply:
Oh - I think we can just get the CB instructiond details from the meta? We have it since the `Tx` here is `TransactionWithMeta`, right?

Reply:
I have updated the function to take fee details. I think you or @tao-stones might be better equipped to refactor it further to use CB instructions from `TransactionWithMeta`.